.github/scripts/strategy-matrix/generate.py

```diff
@@ -238,20 +240,156 @@ def generate_strategy_matrix(all: bool, config: Config) -> list:
         # Add the configuration to the list, with the most unique fields first,
         # so that they are easier to identify in the GitHub Actions UI, as long
         # names get truncated.
+        # Add Address and Thread (both coupled with UB) sanitizers when the distro is bookworm.
+        if os[
+            "distro_version"
+        ] == "bookworm" and f"{os['compiler_name']}-{os['compiler_version']}" in {
+            "gcc-15",
+            "clang-20",
+        }:
+            addSanitizerConfigs(
+                architecture,
+                os,
+                build_type,
+                cmake_args,
+                config_name,
+                build_only,
+                cmake_target,
+                cxx_flags,
+                configurations,
+            )
+        else:
+            if cxx_flags:
+                cmake_args_flags = f"{cmake_args} -DCMAKE_CXX_FLAGS={cxx_flags}"
+            else:
+                cmake_args_flags = f"{cmake_args}"
+            configurations.append(
+                {
+                    "config_name": config_name,
+                    "cmake_args": cmake_args_flags,
+                    "cmake_target": cmake_target,
+                    "build_only": build_only,
+                    "build_type": build_type,
+                    "os": os,
+                    "architecture": architecture,
+                }
+            )
+
+    return configurations
+
+
+def addSanitizerConfigs(
+    architecture: dict,
+    os: dict,
+    build_type: str,
+    cmake_args: str,
+    config_name: str,
+    build_only: bool,
+    cmake_target: str,
+    cxx_flags: str,
+    configurations: list[dict],
```

Comment:
Changing input param is not great.
Let's add a return type here, and in the caller extend the list